xbmc/rendering/MatrixGL.cpp

```diff
@@ -129,6 +129,8 @@ void CMatrixGL::Rotatef(GLfloat angle, GLfloat x, GLfloat y, GLfloat z)
 }
 
 #if defined(HAS_NEON) && !defined(__LP64__)
+#pragma GCC push_options
```

Comment:
please refrain from thinks like this. We are in the process of getting rid of ifdefs and push_option is even more hacky.
move platform code to a separate file and include via cmake

Reply:
will also break on non-gcc toolchains

Reply:
Thx for the reviews
I'm not sure I will be able to turn this file as pure asm .S file which should probably be preferable to a separate file with only the asm volatile and Matrix4Mul function as as cpp.